plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -183,40 +190,58 @@ public InstanceProcess createProcess(Command command, String outputChannel) thro
         return process;
     }
 
-    @Override
     public InstanceKey saveToSnapshot(String owner) throws MachineException {
         try {
             final String repository = generateRepository();
-            String comment = format("Suspended at %1$ta %1$tb %1$td %1$tT %1$tZ %1$tY", System.currentTimeMillis());
-            if (owner != null) {
-                comment = comment + " by " + owner;
+            final String tag = "latest";
+            if (locallySnapshot) {
+                return new DockerInstanceKey(repository, tag, createImage(owner, repository, tag));
+            } else {
+                return new DockerInstanceKey(repository, tag, registry, saveInRegistry(owner, repository, tag));
             }
-            // !! We SHOULD NOT pause container before commit because all execs will fail
-            // to push image to private registry it should be tagged with registry in repo name
-            // https://docs.docker.com/reference/api/docker_remote_api_v1.16/#push-an-image-on-the-registry
-            docker.commit(container, registry + '/' + repository, "latest", comment, owner);
-            //TODO fix this workaround. Docker image is not visible after commit when using swarm
-            Thread.sleep(2000);
-
-            final ProgressLineFormatterImpl progressLineFormatter = new ProgressLineFormatterImpl();
-            String digest = docker.push(repository, "latest", registry, currentProgressStatus -> {
-                try {
-                    outputConsumer.writeLine(progressLineFormatter.format(currentProgressStatus));
-                } catch (IOException ignored) {
-                }
-            });
-
-            docker.removeImage(registry + '/' + repository, false);
-
-            return new DockerInstanceKey(repository, "latest", registry, digest);
-        } catch (IOException e) {
-            throw new MachineException(e);
-        } catch (InterruptedException e) {
+        } catch (IOException ioEx) {
+            throw new MachineException(ioEx);
+        } catch (InterruptedException ex) {
             Thread.currentThread().interrupt();
-            throw new MachineException(e.getLocalizedMessage(), e);
+            throw new MachineException(ex.getLocalizedMessage(), ex);
         }
     }
 
+    @VisibleForTesting
```

Comment:
why VisibleForTesting and not only protected access instead of package access ?
also there is no test calling this method so I don't see why it should be "VisibleForTesting"